frontend/middleware.ts

```diff
@@ -0,0 +1,11 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+import { PROD_USER_AGENT } from '@config/constants';
+
+export function middleware(request: NextRequest) {
+   const init =
+      request.headers.get('user-agent') === PROD_USER_AGENT
+         ? undefined
+         : { headers: { 'X-Robots-Tag': 'noindex, nofollow' } };
+   return NextResponse.next(init);
```

Comment:
Can you link to something that indicates this is how to mutate the response?

The docs only say:

> next() - Returns a NextResponse that will continue the middleware chain 

The docs I've seen don't mention this function taking any arguments.

Reply:
oh oopsies that's not what you're asking

Reply:
Here's the source of `NextResponse.next()`

https://github.com/vercel/next.js/blob/24b600303f2e8074c32b79d9b799ade85772d75e/packages/next/server/web/spec-extension/response.ts#L80-L85

Reply:
Thanks!